pkg/asset/installconfig/azure/session.go

```diff
@@ -224,9 +226,38 @@ func newSessionFromCredentials(cloudEnv azureenv.Environment, credentials *Crede
 		AADEndpoint:  cloudEnv.ActiveDirectoryEndpoint,
 	}
 	c.Resource = cloudEnv.TokenAudience
-	authorizer, err := c.Authorizer()
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to get client credentials authorizer")
+
+	authHost := azidentity.AzurePublicCloud
+
+	var authorizer autorest.Authorizer
+	switch cloudEnv.Name {
+	case azure.USGovernmentCloud.Name():
+		authHost = azidentity.AzureGovernment
+		fallthrough
+	case azure.PublicCloud.Name():
+		options := &azidentity.ClientSecretCredentialOptions{AuthorityHost: authHost}
+		cred, err := azidentity.NewClientSecretCredential(credentials.TenantID, credentials.ClientID, credentials.ClientSecret, options)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get client credentials secret")
+		}
+
+		scopes := []string{endpointToScope(cloudEnv.ResourceManagerEndpoint)}
+		// Use an adapter so azidentity in the Azure SDK can be used as
+		// Authorizer when calling the Azure Management Packages, which we
+		// currently use. Once the Azure SDK clients (found in /sdk) move to
+		// stable, we can update our clients and they will be able to use the
+		// creds directly without the authorizer. The schedule is here:
+		// https://azure.github.io/azure-sdk/releases/latest/index.html#go
+		authorizer = azidext.NewTokenCredentialAdapter(cred, scopes)
+	default:
```

Comment:
This works, but having Azure Stack as a default is counter intuitive because it is a much less common use case. I would like to make this more explicit. Let's go back to the earlier option we discussed of passing `cloudName azure.CloudEnvironment` as a parameter to this function and keep the switch statement logic similar to what it was before.